### PR TITLE
Initial small rewrite of event reducers

### PR DESCRIPTION
**Pull request by odinuge**

### CODE REVIEW

File: app/reducers/events.js [hunk 1]
```diff
@@ -94,9 +100,14 @@ function mutateEvent(state: any, action: any) {
...
               ? stateEvent.registrationCount - 1
               : stateEvent.registrationCount,
-            waitingRegistrations: (
-              stateEvent.waitingRegistrations || []
-            ).filter(id => id !== action.payload.id)
+            waitingRegistrationCount: fromPool
+              ? stateEvent.waitingRegistrationCount
+              : stateEvent.waitingRegistrationCount - 1,
+            waitingRegistrations:
+              stateEvent.waitingRegistrations &&
```
Comment (odinuge) [inline app/reducers/events.js]: This is the best way I can think of (atm.) to keep the behavior somewhat logical. If the event doesn't have `waitingRegistrations` (not defined), we should not "make it" into an array when receiving a socket update.

Comment (odinuge) [reply app/reducers/events.js]: This will still introduce the key `waitingRegistrations` with a value of `undefined` into the event object.

Comment (hanswilw) [reply app/reducers/events.js]: `...(stateEvent.waitingRegistrations && ` can avoid that, not the most readable though

Comment (odinuge) [reply app/reducers/events.js]: Aha @hanswilw.

Didn't think about that. That solution will be even better :smile:

**Comment by odinuge** [comment]
Any thoughts @hanswilw. These reducers are causing a quite a few bugs, so it would be nice to get this into production.

**Pull request merged**